composer/callbacks/system_metrics_monitor.py

```diff
@@ -0,0 +1,75 @@
+# Copyright 2022 MosaicML Composer authors
+# SPDX-License-Identifier: Apache-2.0
+
+"""System metrics monitor callback."""
+
+from __future__ import annotations
+
+import logging
+import os
+
+import psutil
+import pynvml
+
+from composer.core import Callback, Event, State
+from composer.loggers import Logger
+from composer.utils import dist
+
+log = logging.getLogger(__name__)
+
+__all__ = ['SystemMetricsMonitor']
+
+
+class SystemMetricsMonitor(Callback):
+    """Track system metrics."""
+
+    def __init__(self) -> None:
+        super().__init__()
```

Comment:
import it inside the class instead
```suggestion
        super().__init__()
         try:
            import pynvml
         except ImportError as e:
            raise MissingConditionalImportError(extra_deps_group='pynvml',
                                                conda_package='pynvml',
                                                conda_channel='conda-forge') from e
```